internal/shared/shared.go

```diff
@@ -0,0 +1,52 @@
+package shared
+
+import (
+	"context"
+	"sync"
+
+	"golang.org/x/sync/singleflight"
+
+	"github.com/openfga/openfga/internal/cachecontroller"
+	serverconfig "github.com/openfga/openfga/internal/server/config"
+	"github.com/openfga/openfga/pkg/storage"
+)
+
+type SharedResources struct {
```

Comment:
Resources is a broad term, it'd be good to have something indicating that these are for `check` specifically. Maybe `SharedCheckResources`?